### ISSUE TITLE
linux-config-5.15.12 does not build

### ISSUE THREAD

**Issue by legendofmiracles** [NixOS/nixpkgs#153493]
### Describe the bug

```
GOT:
GOT: #
GOT: # configuration written to .config
GOT: #
GOT: make[1]: Leaving directory '/build/linux-5.15.12/build'
GOT: make: Leaving directory '/build/linux-5.15.12'
warning: unused option: BPF_LSM
warning: unused option: EXT4_ENCRYPTION
warning: unused option: F2FS_FS_ENCRYPTION
warning: unused option: KEXEC_FILE
warning: unused option: NUMA
warning: unused option: RCU_PERF_TEST
warning: unused option: SWIOTLB_XEN
error: unused option: X86_SGX
error: unused option: X86_SGX_KVM
warning: unused option: XEN_BALLOON_MEMORY_HOTPLUG
warning: unused option: XEN_DOM0
warning: unused option: XEN_EFI
warning: unused option: XEN_HAVE_PVMMU
warning: unused option: XEN_MCE_LOG
warning: unused option: XEN_SCRUB_PAGES
warning: unused option: XEN_SELFBALLOONING
warning: unused option: XEN_STUB
warning: unused option: XEN_TMEM
```

I'm running recent master, commit `79e19b4361cdb7ab025f10dc195058294f87535f`: "python38Packages.vt-py: 0.11.0 -> 0.12.0".
I have no linux kernel config, other than using the xanmod kernel.

### Steps To Reproduce
1. Use the described commit
2. Try to build linux-config i guess
3. You should see this error, if you don't maybe try to use the xanmod kernel in boot.kernelPackages

### Notify maintainers
https://github.com/NixOS/nixpkgs/pull/147693 might be at fault

@veehaitch @Mic92 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.11-xanmod1-tt, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.0pre20211217_6e6e998`
 - channels(root): `""`
 - channels(nix): `""`
 - nixpkgs: `/nix/store/qynpx9m3aw41m3md91nqp0477fiwf6c8-qzvq1qdksi99hldiqfgr4kk57dha4i41-source`
```

**Comment by astatide** [comment]
Are you running this on an AMD machine?  I'm running into this same issue with _any_ kernel on my Ryzen desktop.  It seems that it's trying to set these config variables (as they're required and have no dependency on vendor, despite being a vendor extension) and... my best guess is that it chokes when the variable is ignored by the actual build process.  I'm not proficient enough at how the kernel is built in nixos to be able to properly verify it, but the derivation for the kernel has X86_SGX and X86_SGX_KVM in the common config options regardless of vendor.

My laptop, which runs an Intel CPU, has zero problem with this.

**Comment by cschramm** [comment]
Judging from other options like `NUMA` and `KEXEC_FILE` in the output it seems like it's not a 64 bit build of the kernel, although `system: "x86_64-linux"` hints at it. Those options do require a 64 bit build, though, so that they do not get read in that case. The condition for the SGX options is fixed in #153525, maybe that helps ... whatever the reason for the 32 bit build might be.

**Comment by legendofmiracles** [comment]
I've updated to `22.05-pre-10185-gb7018f38133` and now everything seems to work? Despite when I tried https://github.com/NixOS/nixpkgs/pull/153525 it errored out with something else.
I'm honestly not sure why this happened, I'm running a 64 bit system so it should only be building that kernel. 

I'm running an Intel CPU, so that also wasn't it @astatide. If it works for you now feel free to close
